### PR TITLE
core: add admin restriction to environments route

### PR DESCRIPTION
**Pull request by BrianJayChung**
temporary fix to disallow all environments to be able to access the `/environments` route.

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -240,16 +240,20 @@ def upgrade():
...
-            project_pick = pickle.dumps(project)
-            current_app.redis_client.set(PROJECT_NAME, project_pick)
-            return jsonify({'response': 200})
-        except Exception as e:
-            current_app.logger.error(e)
-            return jsonify({'response': 400})
+    url = url_parse(request.url)
+    subdomain = url.host.split('.')[0]
+    if subdomain == 'admin':
+        if webhook:
```
Comment (InTheCloudDan) [inline app/routes.py]: @BrianJayChung can we merge these 2 if statements together using `and`?

File: app/routes.py [hunk 2]
```diff
@@ -240,16 +240,20 @@ def upgrade():
...
+                current_app.redis_client.set(PROJECT_NAME, project_pick)
+                return jsonify({'response': 200})
+            except Exception as e:
+                current_app.logger.error(e)
+                return jsonify({'response': 400})
+        else:
+            abort(404)
     else:
-        abort(404)
+        return "NO ACCESS"
```
Comment (InTheCloudDan) [inline app/routes.py]: this should return the proper http status of unauthorized. I know I also need to fix jsonify to do it for the webhook itself.

File: app/routes.py (hunk 1 above)
Comment (BrianJayChung) [reply app/routes.py]: If they are merged how should we track between domain not matching and flag evaluated, since they will both hit abort(404)

Comment (InTheCloudDan) [reply app/routes.py]: do we generally need to discern between the differences?

Comment (BrianJayChung) [reply app/routes.py]: Not really, since this is really only meant to be accessed by a few.

**Comment by BrianJayChung** [comment]
@InTheCloudDan Let me know if you have something else in mind.

**Pull request merged**